clients/js/evm.ts

```diff
@@ -59,23 +82,51 @@ export async function query_contract_evm(
       }
       const tb = BridgeImplementation__factory.connect(contract_address, provider)
       result.address = contract_address
-      result.wormhole = await tb.wormhole()
-      result.implementation = (await getStorageAt(rpc, contract_address, _IMPLEMENTATION_SLOT, ["address"]))[0]
+      const registrationsPromise = Promise.all(
+        Object.entries(CHAINS)
+          .filter(([c_name, _]) => c_name !== chain && c_name !== "unset")
+          .map(async ([c_name, c_id]) => [c_name, await tb.bridgeContracts(c_id)])
+      )
+      let [
+        wormhole,
+        implementationSlotTb,
+        tokenImplementation,
+        chainIdTb,
+        finality,
+        evmChainIdTb,
+        isForkTb,
+        governanceChainIdTb,
+        governanceContractTb,
+        WETH,
+        registrations
+      ] = await Promise.all([
+        tb.wormhole(),
+        getStorageAt(rpc, contract_address, _IMPLEMENTATION_SLOT, ["address"]),
+        tb.tokenImplementation(),
+        tb.chainId(),
+        tb.finality(),
+        maybeUnsupported(tb.evmChainId()),
+        maybeUnsupported(tb.isFork()),
+        tb.governanceChainId(),
+        tb.governanceContract(),
+        tb.WETH(),
+        registrationsPromise
+      ])
+      result.wormhole = wormhole
```

Comment:
Would be useful for legibility and correctness to type `result`. Could declare as `Partial<ResultType>` so these incremental assignments don't error and assert `ResultType` before returning.

Reply:
good idea!

Reply:
will merge this is now, I want to do a general cleanup of the client code in a separate PR